src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterChar.cs

```diff
@@ -8,12 +8,12 @@ namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class JsonConverterChar : JsonConverter<char>
     {
-        public override char Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        public override char Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions? options)
         {
-            return reader.GetString()[0];
+            return reader.GetString()![0];
```

Comment:
This is a nullref bug.

This should be implemented as follows:
```C#
public override char Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
{
    string? str = reader.GetString();
    if (string.IsNullOrEmpty(str) || str.Length > 1)
    {
        throw ThrowHelper.GetInvalidOperationException_ExpectedChar(reader.TokenType);
    }
    return str[0];
}

// In ThrowHelper.cs
public static InvalidOperationException GetInvalidOperationException_ExpectedChar(JsonTokenType tokenType)
{
    return GetInvalidOperationException("char", tokenType);
}
```

Here are some tests that we should add:
```C#
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("null"));
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>(""));
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("1234"));
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"stringTooLong\""));
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"\u0059\"B"));
Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"\uD800\uDC00\""));
Assert.Equal('a', JsonSerializer.Deserialize<char>("\"a\""));
Assert.Equal('Y', JsonSerializer.Deserialize<char>("\"\u0059\""));
```

Reply:
Good catch!